### ISSUE TITLE
SPO requirements?

### ISSUE THREAD

**Issue by ahwagner** [ga4gh/va-spec#74]
@mbrush says it is imperative to retain SPO structure in propositions explicitly, not by slot inheritance.

@larrybabb mentioned we need to make qualifiers clearer, @mbrush mentioned there are typed qualifiers and referenced previous documentation (@mbrush please point us at that here).

**Comment by mbrush** [comment]
As I noted on the call, I think expressing the meaning of a Proposition using the SPOQ structure ( subject, predicate, object, and optional qualifier(s)) provides a unified way to read what is being asserted in a Statement, that is clear and consistent across all Statement/Proposition types. The SPOQ paradigm has been established for years as a foundational and ubiquitous component of VA-SEPIO models, and our documentation will prime users of our models or data to recognize the intent and utility of this pattern.  

As for an example of 'specialized' qualifies defined for specific 'profiles' - see the various qualifier properties defined for the draft [Therapeutic Efficacy Statement profile here](https://docs.google.com/spreadsheets/d/1zQU-Yv7gB7IHKIOVsTh-74BwdtgB9KQpKcWkSHZOa-Q/edit#gid=1646330759&range=118:118)  (rows 130-133 hold specialized qualifier attributes at the time of this writing - each clearly defining the semantics of the data it holds).  These more specific qualifiers are defined as part of 'profiles', which specialize the domain agnostic core IM to represent a specific type of Statement,.

**Comment by ahwagner** [comment]
@larrybabb and I reviewed the specialized qualifiers described and think that the current abstract Proposition (explicit SPO slots) with concrete subclasses (inherited & extended SPO slots) is conceptually aligned to the notion of "specialized qualifiers" as shown in the above document.

We think this is better aligned to the way the associated schemas are structured and fully parsable as SPO via JSON-LD. We are open to revisiting this in the future but for now are focused on an implementation by plenary and think this the most consistent path forward.

**Comment by mbrush** [comment]
Adding Larry's related comment from the original [ga4gh-va-model spreadsheet ](https://docs.google.com/spreadsheets/d/1zQU-Yv7gB7IHKIOVsTh-74BwdtgB9KQpKcWkSHZOa-Q/edit#gid=1646330759&range=A130) - as it is relevant to this discussion:

> @matt@tislab.org - Can you clarify whether the variety of "qualifier" attributes in this schema will actually be "renamed" to the more specialized term in the parentheses (e.g. disease_context_qualifier')?  If so, then why not also be able to specialize the "subject" and "object" to be more precisely named to the type of content they have. If not, then how would one know which "qualifier" items go with which "qualifier" attributes in your list?

**Comment by mbrush** [comment]
Adding my response to the questions and proposal to drop SPO labels from these slots in the implementation model (apologies in advance for its length - but I feel strongly about this ;-):

First, to address questions about **qualifier specialization** (as this seems to be a key justification/rationale for also specializing SPO slots).  
- The qualifier slot is a different case, because unlike SPO slots, a given Statement may have multiple qualifiers - each extending or refining the core SPO triple with a different type of information (e.g. see the TherapeuticEfficacyStatement Profile [here](https://docs.google.com/spreadsheets/d/1zQU-Yv7gB7IHKIOVsTh-74BwdtgB9KQpKcWkSHZOa-Q/edit#gid=1646330759&range=A118)).   
- It is for this reason that we permit specialization of the core 'qualifier' slot - so that we can differentiate between various qualifiers that may be applied to a given SPO statement.  
- Defining qualifiers with specific semantics also helps users understand the role the qualifier plays, and how the information it holds is related the core SPO triple.  
- This is not needed for S and O slots, because the predicate defines the relationship between them.  We name / define qualifiers in such a way as to be clear about how the info they hold relates to the SPO.
- Finally, note that when we do specialize the core qualifier slot, we keep 'qualifier' at the end of the label so it is clear that this slot holds information that qualifies the SPO (e.g. `disease_context_qualifier`).  If you insist on specializing SPO slots (which I hope I can persuade you against) - consider doing the same (e.g. `subject_variation` or `variation_subject`, and `object_condition` or `condition_object`)
- If we drop explicit reference to SPOQ in the slots that capture statement semantics - we abandon a core principle of the VA-SEPIO model that aspires to make semantics of what a statement or proposition expresses explicit, unambiguous and consistent across all types of knowledge.

--------

TL;DR - The rest of the text below reiterates the case I have made many times for maintaining explicit SPO slots in VA profiles / implementation models - and more generally lays out my philosophy and approach for the profiling paradigm.  You've likely heard it before, but a refresher can't hurt.
- I feel so strongly about the explicit use of SPO slots because this has been a foundational tenet of the VA and SEPIO approach from the outset. It is IMO critical to the primary goal of providing an ecosystem of Statements where the core knowledge they express is conveyed in an unambiguous, computable, and consistent way across any type of statement, in any system or implementation . . . so anyone can grasp what a Statement is asserting  whether they are an inside or outsider. The immediately apparent SPOQ pattern is the unifying thread across all VA SEPIO data. It is at the heart of what makes SEPIO SEPIO.
 
- I recognize the motivation to use slot names that your established core users expect to see - and as an implementation model this is completely understandable.  But I would argue that in the larger landscape / context in which your data may be viewed -  by users of diverse expertise and familiarity - there are greater benefits to providing a consistent and explicit interface to statement semantics through use of the SPOQ pattern.  This may not be a central concern if you intend your profile to only be used within ClinGen and VICC systems - and will transform to a VA standard format that uses the core SPO slots.  But even so - uniform adherence to the SPOQ pattern in all profiles will  go a long way to ingrain this paradigm in the community of use so it becomes ubiquitous and intuitive. This is harder to achieve if we hide SPOQ semantics under layers of inheritance/abstraction in the model.  And changing slot names to suit community preference will end up making these models unrecognizable as VA-SEPIO profiles.  
- IMO, the further upstream we can push the core concepts and terminology of the VA-SEPIO framework, the better chance we have at success.  This may be overambitious - but I think it would be hugely beneficial to have implementations 'speaking SEPIO' in their models as much as possible.   And we will lay the foundation for this paradigm in excellent primers and onboarding documentation / videos - so that users approach models and data from a common and informed perspective.
- In closing - I fear we will end up with a landscape of models/profiles that have no clear resemblance to each other, and in which the underlying standard on which they are based is unrecognizable. I know we said the flexibility to customize to a given application or implementation is important for adoption, but worry that we may go too far in that direction.  Finally, from a community outreach perspective, if the CrC model you are developing will be the first glimpse the world gets of a VA Profile  - it would be nice if the core bits of the foundational VA model are easily recognizable.

OK, I'll get off my soap box now. We can revisit when we talk next if there are questions, and for my part I’d like to better understand what specific concerns you have about use of the explicit SPO slots as they in your implementation model.
